crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -107,16 +107,28 @@ impl Solution {
             .sorted()
     }
 
+    /// Verify that the solution is valid and can be broadcast safely. See
+    /// [`settlement::Verified`].
+    pub async fn verify(
+        &self,
+        eth: &Ethereum,
+        simulator: &Simulator,
+        auction: &competition::Auction,
+    ) -> Result<settlement::Verified, Error> {
+        self.verify_asset_flow()?;
+        self.verify_internalization(auction)?;
+        self.simulate(eth, simulator, auction).await
```

Comment:
According to the [issue](https://github.com/cowprotocol/services/issues/1009#issuecomment-1387621803) both the simulations for the internalized as well as the uninternalized version need to pass. It's fine to do it in a separate PR; just wanted to point it out so we don't forget about it.